### DISCUSSION TITLE
Reload nginx when add conf

### DISCUSSION THREAD

**Discussion by skillcrushbinguyen** [nginx-proxy/nginx-proxy#1782] [General]
Hi everybody,
i want to known how to detect custom conf file change then reload nginx. As far as I know, this container **nginx-proxy** only reloads when there is a change from the docker container.

**Comment by buchdag** [comment]
Hi. I think the only way you'd be able to do that at the moment would be to send SIGHUP to the docker-gen process to trigger a re-render.

**Comment by Dr-Shadow** [comment]
For anyone having issues with reload, you can signal the process you need with these commands :

```
docker exec -it nginx-proxy sh -c 'kill -SIGHUP $(pidof docker-gen)'
docker exec -it nginx-proxy sh -c 'kill -SIGHUP $(cat /run/nginx.pid)'
```

Of course, you will probably need to escape the $ or ' depending on your use case (in scripts, etc...)
